unit-tests/run-unit-tests.py

```diff
@@ -197,34 +229,59 @@ def cat( filename ):
     except FileNotFoundError:
         error( red + testname + reset + ': executable not found! (' + exe + ')' )
         continue
+    
+    check_log(log, testname, exe)
 
-    # Normal logs are expected to have in last line:
-    #     "All tests passed (11 assertions in 1 test case)"
-    # Tests that have failures, however, will show:
-    #     "test cases: 1 | 1 failed
-    #      assertions: 9 | 6 passed | 3 failed"
-    for ctx in grep( r'^test cases:\s*(\d+) \|\s*(\d+) (passed|failed)', log ):
-        m = ctx['match']
-        total = int(m.group(1))
-        passed = int(m.group(2))
-        if m.group(3) == 'failed':
-            # "test cases: 1 | 1 failed"
-            passed = total - passed
-        if passed < total:
-            if total == 1  or  passed == 0:
-                desc = 'failed'
-            else:
-                desc = str(total - passed) + ' of ' + str(total) + ' failed'
 
-            if verbose:
-                error( red + testname + reset + ': ' + desc )
-                info( 'Executable:', exe )
-                info( 'Log: >>>' )
-                out()
-                cat( log )
-                out( '<<<' )
-            else:
-                error( red + testname + reset + ': ' + desc + '; see ' + log )
+current_dir = os.path.dirname(os.path.abspath(__file__))
+current_dir = current_dir.replace('\\' , '/')
+# this script is located in librealsense/unit-tests, so one directory up is the main repository
+librealsense = os.path.dirname(current_dir)
+
+# Python scripts should be able to find the pyrealsense2 .pyd or else they won't work. We don't know
+# if the user (Travis included) has pyrealsense2 installed but even if so, we want to use the one we compiled.
+# we search the librealsense repository for the .pyd file (.so file in linux)
+pyrs = ""
+if linux:
+    for so in find(librealsense, '(^|/)pyrealsense2.*\.so$'):
+        pyrs = so
+else:
+    for pyd in find(librealsense, '(^|/)pyrealsense2.*\.pyd$'):
+        pyrs = pyd
+# if we run python tests with no .pyd/.so file they will crash. Therefore we only run them if such a file was found
+if pyrs:
+    pyrs_path = librealsense + '/' + pyrs
```

Comment:
Use `os.sep`?

Reply:
I previously changed all the back-slashes to slashes so it seems strange to not use a slash here as well. I did this because slashes are compatible to both os.
Would you prefer I change it to using os.sep? here and in the previous place